### PR TITLE
fix(stableswap): require sorted initial liquidity

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #XXX

## What is the purpose of the change

Currently, the CLI auto-sorts the initial liquidity. As a result, denoms end up mismatching the desired scaling factor.

This PR  removes the sorting in CLI and adds the requirement in `ValidateBasic()` for initial liquidity coins to be sorted. Fails if not.

## Testing and Verifying

Tested on localnet. Summary: https://hackmd.io/40cR4cL5SBOuX7APZKTWIw?edit

### CODE REVIEW

File: x/gamm/client/cli/tx.go [hunk 1]
```diff
@@ -855,3 +855,17 @@ func NewStableSwapAdjustScalingFactorsMsg(clientCtx client.Context, txf tx.Facto
 
 	return txf, msg, nil
 }
+
+func ParseCoinsNoSort(coinsStr string) (sdk.Coins, error) {
```
Comment (p0mvn) [inline x/gamm/client/cli/tx.go:859]:
```suggestion
// ParseCoinsNoSort parses coins from coinsStr but does not sort them.
// Returns error if parsing fails.
func ParseCoinsNoSort(coinsStr string) (sdk.Coins, error) {
```

File: x/gamm/pool-models/stableswap/pool.go [hunk 2]
```diff
@@ -418,13 +418,22 @@ func validateScalingFactors(scalingFactors []uint64, numAssets int) error {
 
 // assumes liquidity is all pool liquidity, in correct sorted order
 func validatePoolLiquidity(liquidity sdk.Coins, scalingFactors []uint64) error {
+	liquidityCopy := make(sdk.Coins, len(liquidity))
+	copy(liquidityCopy, liquidity)
+	liquidityCopy.Sort()
+
```
Comment (ValarDragon) [inline x/gamm/pool-models/stableswap/pool.go:424]: Can we add a test for the go logic?

**Review by ValarDragon — APPROVED** [review]
LGTM, I think we should add a test for the go logic though

File: x/gamm/pool-models/stableswap/pool.go [hunk 3]
```diff
@@ -418,13 +418,22 @@ func validateScalingFactors(scalingFactors []uint64, numAssets int) error {
 
 // assumes liquidity is all pool liquidity, in correct sorted order
 func validatePoolLiquidity(liquidity sdk.Coins, scalingFactors []uint64) error {
+	liquidityCopy := make(sdk.Coins, len(liquidity))
```
Comment (alexanderbez) [inline x/gamm/pool-models/stableswap/pool.go:421]: Any reason `sdk.NewCoins(liquidity...)` would work?

**Review by alexanderbez — APPROVED** [review]

File: x/gamm/pool-models/stableswap/pool.go (hunk 3 above)
Comment (ValarDragon) [reply x/gamm/pool-models/stableswap/pool.go:421]: agreed, that seems better to me as well

Comment (p0mvn) [reply x/gamm/pool-models/stableswap/pool.go]: It also sorts the input coins. We don't want that

As a result, we would have to make a new copy anyways

**Pull request merged**